modules/core/04-channel/types/msgs_test.go

```diff
@@ -518,3 +518,96 @@ func (suite *TypesTestSuite) TestMsgChannelUpgradeCancelGetSigners() {
 	msg := types.NewMsgChannelUpgradeCancel(ibctesting.MockPort, ibctesting.FirstChannelID, types.ErrorReceipt{Sequence: 1}, suite.proof, height, addr)
 	suite.Require().Equal([]sdk.AccAddress{expSigner}, msg.GetSigners())
 }
+
+func (suite *TypesTestSuite) TestMsgChannelUpgradeConfirmValidateBasic() {
+	var msg *types.MsgChannelUpgradeConfirm
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {},
+			true,
+		},
+		{
+			"invalid port identifier",
+			func() {
+				msg.PortId = invalidPort
+			},
+			false,
+		},
+		{
+			"invalid channel identifier",
+			func() {
+				msg.ChannelId = invalidChannel
+			},
+			false,
+		},
+		{
+			"cannot submit an empty channel proof",
+			func() {
+				msg.ProofChannel = emptyProof
+			},
+			false,
+		},
+		{
+			"cannot submit an empty upgrade error proof",
+			func() {
+				msg.ProofUpgradeError = emptyProof
+			},
+			false,
+		},
+		{
+			"cannot submit an empty upgrade sequence proof",
+			func() {
+				msg.ProofUpgradeSequence = emptyProof
+			},
+			false,
+		},
+		{
+			"proof height must be > 0",
+			func() {
+				msg.ProofHeight = clienttypes.ZeroHeight()
+			},
+			false,
+		},
+		{
+			"invalid counterparty channel state",
+			func() {
+				msg.CounterpartyChannel.State = types.TRYUPGRADE
```

Comment:
nit: feel like this case should be above the proof cases

Reply:
done